### DISCUSSION TITLE
Preemptive authentication

### DISCUSSION THREAD

**Discussion by Joakim-T** [thunderclient/thunder-client-support#1300] [Q&A]
Struggling authenticating with a REST API that requires preemptive authentication.
How do i configure this in Thunder Client?

**Comment by rangav** [comment]
Hi @Joakim-T 

Can you share more info like code samples or screenshots

**Reply by Joakim-T** [reply]
Hello!
Basic authentication (Dummy user/password below) and i get 401 Unauthorized back.
A colleague uses SOAP UI and the same request works when he enables "Authenticate pre-emptively".
Is this somehing Thunder Client cannot do?

```
var client = new HttpClient();
var request = new HttpRequestMessage();
request.RequestUri = new Uri("https://test/api/query/org?companyID=AV&select=avd");
request.Method = HttpMethod.Get;

request.Headers.Add("Accept", "application/json");
request.Headers.Add("User-Agent", "Thunder Client");
request.Headers.Add("Authorization", "Basic xyz");

var response = await client.SendAsync(request);
var result = await response.Content.ReadAsStringAsync();
Console.WriteLine(result);
```

**Reply by rangav** [reply]
The code sample you shared looks like a simple GET request with base auth. I am not sure what SOAP UI is doing additionally.

Will do some research and get back to you.

**Reply by rangav** [reply]
preemptive authentication?

`This means that the browser is not authenticated with the initial request, but rather gets an 'unauthorized' response for the server, which triggers the authentication request to be sent after. If the server does not respond with a 400 error, the authentication is not sent after and the test will error.`

https://community.smartbear.com/t5/AlertSite-Questions/Preemptive-Authentication-and-SoapUI-Tests-versus-Single-URL/m-p/157896/highlight/true#M355

**Reply by gambrjl** [reply]
Has anyone found a work around for this?  from the link above the API I'm calling is giving 403 Unauthorized and the auth credentials are not being sent b/c it's not 400 if I'm understanding.  SOAP works with the Authenticate pre-emptively flag
